onnx/defs/math/defs.cc

```diff
@@ -2527,40 +2545,42 @@ bool BuildContextDependentFunctionBodySCE(
     const OpSchema& schema,
     FunctionProto& functionProto) {
   std::vector<FunctionBodyHelper::NodeDef> body;
-  body.push_back({{"X_Max"}, "Max", {"scores"}});
+
+  body.push_back({{"X_Max"},
+                  "ReduceMax",
+                  {"scores"},
+                  {MakeAttribute("axes", std::vector<int64_t>({1}))}});
+
   body.push_back({{"X_Sub"}, "Sub", {"scores", "X_Max"}});
   body.push_back({{"X_Exp"}, "Exp", {"X_Sub"}});
-  body.push_back({{"X_RS"}, "ReduceSum", {"X_Exp"}});
+  body.push_back({{"X_RS"}, "ReduceSum", {"X_Exp"}, {MakeAttribute("axes", std::vector<int64_t>({1}))}});
   body.push_back({{"X_Div"}, "Div", {"X_Exp", "X_RS"}});
-  body.push_back({{"log_prob"}, "Log", {"X_Div"}});
-  if (ctx.getAttribute("ignore_index") == nullptr) {
-    if (!ctx.hasInput(2)) {
-      body.push_back({{"output"},
-                      "NegativeLogLikelihoodLoss",
-                      {"log_prob", "labels"},
-                      {MakeRefAttribute("reduction", AttributeProto::STRING)}});
-    } else {
-      body.push_back({{"output"},
-                      "NegativeLogLikelihoodLoss",
-                      {"log_prob", "labels", "weights"},
-                      {MakeRefAttribute("reduction", AttributeProto::STRING)}});
-    }
-  } else {
-    if (!ctx.hasInput(2)) {
-      body.push_back({{"output"},
-                      "NegativeLogLikelihoodLoss",
-                      {"log_prob", "labels"},
-                      {MakeRefAttribute("reduction", AttributeProto::STRING),
-                       MakeRefAttribute("ignore_index", AttributeProto::INT)}});
-    } else {
-      body.push_back({{"output"},
-                      "NegativeLogLikelihoodLoss",
-                      {"log_prob", "labels", "weights"},
-                      {MakeRefAttribute("reduction", AttributeProto::STRING),
-                       MakeRefAttribute("ignore_index", AttributeProto::INT)}});
-    }
+  body.push_back({{"X_Log"}, "Log", {"X_Div"}});
+
+  // Review(mzs): Ideally we want to reuse the output from Log for sub-graph output as well but
+  // looking at the graph resolve code it does not include graph outputs as intermediate outputs, hence
+  // if intermediate X_log is renamed as log_prob then it will be treated as graph output and will not 
+  // be available to NegativeLogLikelihoodLoss. May be my understanding is incorrect or there is a bug in
+  // function population code in ORTbut I will dig further to be 100%.
```

Comment:
I suspect it might be a bug in ORT function expansion code. E.g., something similar to what is done in ONNX might be required: see: https://github.com/onnx/onnx/blob/2dd6a1ba5fcdb1d47cd485d5033e6cf804100d49/onnx/defs/function.cc#L46

Reply:
Agreed, I will track this separately, spoke offline and @gramalingam is fine with the workaround for now.